bundles/io/org.eclipse.smarthome.io.net/src/main/java/org/eclipse/smarthome/io/net/http/HttpRequestBuilder.java

```diff
@@ -0,0 +1,143 @@
+    }
+
+    /**
+     * Construct an http request builder to get data from an <code>url</code>
+     *
+     * @param url to fetch the data from
+     * @return a request builder to construct and complete the request
+     */
+    public static HttpRequestBuilder getFrom(String url) {
+        return new HttpRequestBuilder("GET", url);
+    }
+
+    /**
+     * Construct an http request builder to post data to a <code>url</code>
+     *
+     * @param url to post data to
+     * @return a request builder to construct and complete the request
+     */
+    public static HttpRequestBuilder postTo(String url) {
+        return new HttpRequestBuilder("POST", url);
+    }
+
+    /**
+     * Add content to this request
+     *
+     * @param content a string containing the data to be pushed to the <code>url</code>
+     * @return a request builder to construct and complete the request
+     */
+    public HttpRequestBuilder withContent(String content) {
+        this.inputStream = new ByteArrayInputStream(content.getBytes(StandardCharsets.UTF_8));
+        return this;
+    }
+
+    /**
+     * Add content with a specific type to this request
+     *
+     * @param content a string containing the data to be pushed to the <code>url</code>
+     * @param contentType the content type of the given <code>content</code>
+     * @return a request builder to construct and complete the request
+     */
+    public HttpRequestBuilder withContent(String content, String contentType) {
+        withContent(content);
+        this.contentType = contentType;
+        return this;
+    }
+
+    /**
+     * Add an additional header to the request
+     *
+     * @param header name of the header, eg. Content-Type
+     * @param value value of the header, eg. "application/json"
+     * @return a request builder to construct and complete the request
+     */
+    public HttpRequestBuilder withHeader(String header, String value) {
+        Properties localHeaders = headers;
+
+        if (localHeaders == null) {
+            headers = new Properties();
+            localHeaders = headers;
+        }
+
+        localHeaders.put(header, value);
+        return this;
+    }
+
+    /**
+     * Add a timeout for this request
+     *
+     * @param timeout the timeout for this http request as a <code>Duration</code>
+     * @return a request builder to construct and complete the request
+     */
+    public HttpRequestBuilder withTimeout(Duration timeout) {
+        this.timeout = timeout;
+        return this;
+    }
+
+    /**
+     * Executes the build request
+     *
+     * @return the response body or <code>null</code> when the request went wrong
+     * @throws IOException when the request execution failed, timed out or it was interrupted
+     */
+    public String getContentAsString() throws IOException {
+        // this cast should be okay, a requested timeout should normally not exceed a long
+        int timeoutMillis = (int) timeout.toMillis();
```

Comment:
Why not using an `int` or `long` instead of `Duration` for the timeout if you need an `int` here?

Reply:
Because a long without any unit misses a lot of expression power, and there are a lot cases where it would make more sense to express it in seconds or in strange corner cases maybe in minutes